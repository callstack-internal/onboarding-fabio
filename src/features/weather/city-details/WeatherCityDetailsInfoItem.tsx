import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import Colors from '../../../styles/Colors';

type Props = {
  label: string;
  value: string;
};

const WeatherCityDetailsInfoItem = ({label, value}: Props) => {
  return (
    <SafeAreaView style={styles.rootContainer} edges={['left', 'right']}>
      <Text style={styles.label} numberOfLines={1}>
        {label}
      </Text>
      <Text style={styles.valueLabel} numberOfLines={1}>
        {value}
      </Text>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  rootContainer: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: Colors.white,
    padding: 20,
  },
  label: {
    flex: 1,
    fontSize: 14,
    color: Colors.black,
  },
  valueLabel: {
    fontSize: 14,
    color: Colors.gray,
  },
});

export default WeatherCityDetailsInfoItem;
