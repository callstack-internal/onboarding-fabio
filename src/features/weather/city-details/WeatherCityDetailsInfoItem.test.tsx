import React from 'react';
import renderer from 'react-test-renderer';
import WeatherCityDetailsInfoItem from './WeatherCityDetailsInfoItem';

test('should render correctly', () => {
  const tree = renderer
    .create(<WeatherCityDetailsInfoItem label={'Cloud Cover'} value={'50%'} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('should render a different state correctly', () => {
  expect(
    <WeatherCityDetailsInfoItem label={'Cloud Cover'} value={'50%'} />,
  ).toMatchDiffSnapshot(
    <WeatherCityDetailsInfoItem label={'Pressure'} value={'1000 hPa'} />,
  );
});
