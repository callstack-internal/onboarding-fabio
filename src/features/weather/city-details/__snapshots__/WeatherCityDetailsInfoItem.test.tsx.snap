// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a different state correctly 1`] = `
"Snapshot Diff:
- <WeatherCityDetailsInfoItem label="Cloud Cover" value="50%" />
+ <WeatherCityDetailsInfoItem label="Pressure" value="1000 hPa" />

@@ -22,19 +22,19 @@
          "flex": 1,
          "fontSize": 14,
        }
      }
    >
-     Cloud Cover
+     Pressure
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#808080",
          "fontSize": 14,
        }
      }
    >
-     50%
+     1000 hPa
    </Text>
  </RNCSafeAreaView>"
`;
